src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java

```diff
@@ -116,11 +115,24 @@ public ScheduledInterview entityToScheduledInterview(Entity scheduledInterviewEn
   /** Creates a scheduledInterview Entity from a scheduledInterview object. */
   public Entity scheduledInterviewToEntity(ScheduledInterview scheduledInterview) {
     Entity scheduledInterviewEntity = new Entity("ScheduledInterview");
+    setEntityProperties(scheduledInterviewEntity, scheduledInterview);
+    return scheduledInterviewEntity;
+  }
+
+  /** Updates a scheduledInterview Entity from a scheduledInterview object. */
+  public Entity scheduledInterviewToUpdateEntity(ScheduledInterview scheduledInterview) {
+    Entity scheduledInterviewEntity = new Entity("ScheduledInterview", scheduledInterview.id());
+    setEntityProperties(scheduledInterviewEntity, scheduledInterview);
+    return scheduledInterviewEntity;
+  }
+
+  /** Sets the properties of a scheduledInterview Entity. */
+  static void setEntityProperties(
+      Entity scheduledInterviewEntity, ScheduledInterview scheduledInterview) {
```

Comment:
This function goes against something Alex warned me against. That is, a function that modifies a param instead of returning it. Here it might be fine because it's more explicit in the function name but not sure. We can come back after we get some more info.

Reply:
So the conclusion is to put this code back within the two methods because creation of the Entity, property setting, and returning should all be within the same method.

Reply:
Done